### ISSUE TITLE
Port project to OSX

### ISSUE THREAD

**Issue by carlesgutierrez** [carlesgutierrez/ofxSensors4Games#1]
Almost all libraries should be compatible in OSX.  Check best way to do cross-Platform.

**Comment by carlesgutierrez** [comment]
All addons should be portable into OSX. 
Let's use [**Project Generator**](http://openframeworks.cc/learning/01_basics/create_a_new_project/) to add all them and update the project into OSX!

For non cross-platform addons, like **Spout** (that is windows), let's add a variable to manage it like **shareTextureCameraInput** or similar that can use Spout or Syphon( the mac OSX version ), definers will manage to switch between WIN and OSX

**Examples using platform definers:**

#ifdef TARGET_WIN32
//our includes, variables and functions Windows
#endif

 #ifdef TARGET_OSX
//our includes, variables and functions OSX
#endif

#if defined( TARGET_LINUX ) && defined (OF_USING_GTK)
//our includes, variables and functions Linux
#endif

**//More Define examples extracted from [ofSystemUtils]**(https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworks/utils/ofSystemUtils.cpp)
   #elif defined(TARGET_OSX)
      #ifndef MAC_OS_X_VERSION_10_7
          #define OF_VIDEO_PLAYER_QUICKTIME
      #elseif !defined(MAC_OS_X_VERSION_10_8)
          #define OF_VIDEO_PLAYER_QTKIT
     #else
         #define OF_VIDEO_PLAYER_AVFOUNDATION
     #endif

**Comment by omarojo** [comment]
- First I had to clone openFrameworks directly from github. If I use the package from the oficial website for OSX, doesnt work because is missing a bunch of libraries.
- Then I downloaded all the plugins, except for Spout.
- Then I had to run the 'download_libs.sh" located in the scripts folder for OSX, this downloaded a bunch of things into /libs, and move some other into /addons. Some weird stuff right there, This was required otherwise the other libraries will complain about not finding certain other files.
- Then I tried to created a new project using the project generator, Didnt work, it wouldnt allow me, it would show an error about stdout maxBuffer. So I had to compile the projectGenerator project located in /apps, and run it from terminal... in order to generate my project with all the plugins. 
- Then after all the reading in forums...  I was hoping it would finally compile. Nope... now it is complaining with ofxCv.  :( :( 

I posted my issue here: https://github.com/kylemcdonald/ofxCv/issues/244

This is as far as I was able to go.

**Comment by carlesgutierrez** [comment]
Did you download the ofxCv MASTER branch? The one that has to be used ---> STABLE branch!
At Readme you can check this ofxCv tip. 

I've tested on Mac OSX 10.11.6 ( Capitan ) and everything seems to work 

pd: Thanks for all this TIPS OSX ( good to know all steps from scratch for OSX)

**Comment by carlesgutierrez** [comment]
Tell me if it works for you too. 
For the next Update at main example I've to fix this line: 

`mySensorController.setup(cameraSensor, realTimeMode);`

to this: 
`mySensorController.setup(cameraSensor);`

Since last version I could change from Videoplayer to VideoInput directly from Gui

**Comment by carlesgutierrez** [comment]
And I've started to Add some Defines in other to allow to share Texture with Spout and Syphon.  @omarojo If you have suggestions, they are welcome.

For now , please comment this line at [sensorManage.h](https://github.com/carlesgutierrez/ofxSensors4Games/blob/master/src/SensorManager.h) in order to do not use **sharingCameraInput** with Spout. I will update this ASAP

`#define USE_SHARECAM_SPOUT`

**Comment by carlesgutierrez** [comment]
ok, let's work in this branch -->[OSX BRANCH ](https://github.com/carlesgutierrez/ofxSensors4Games/tree/osxPort) There last 2 modifications I've commented here are done. 
We can keep working here until is properly solved.

**Comment by omarojo** [comment]
Yeah I actually read the ofxCV stating I need the STABLE branch.. so yes.. I switched to that from the beginning.

**Comment by omarojo** [comment]
I need to first solve the other issues, before doing the modifications in your project. :(  
Im running macOS 10.13.3 High Sierra with Xcode 9.2
